app/src/main/java/com/framgia/forder/screen/mainpage/MainPageViewModel.java

```diff
@@ -24,26 +28,36 @@
         BaseRecyclerViewAdapter.OnRecyclerViewItemClickListener<Object>, OrderListener {
     private static final String TAG = "ListProductFragment";
 
-    private Navigator mNavigator;
+    private final Navigator mNavigator;
     private MainPageContract.Presenter mPresenter;
-    private ProductAdapter mProductAdapter;
-    private ShopAdapter mShopAdapter;
+    private final ProductAdapter mProductAdapter;
+    private final ShopAdapter mShopAdapter;
+    private final CategoryAdapter mCategoryAdapter;
+    public final ObservableField<Integer> mProgressBarVisibilityProduct = new ObservableField<>();
+    public final ObservableField<Integer> mProgressBarVisibilityShop = new ObservableField<>();
+    public final ObservableField<Integer> mProgressBarVisibilityCategory = new ObservableField<>();
```

Comment:
WARNING: Variable 'mProgressBarVisibilityCategory' must be private and have accessor methods.